graylog2-web-interface/src/views/components/sidebar/SideBar.jsx

```diff
@@ -98,85 +152,60 @@ class SideBar extends React.Component<Props, State> {
-                <span data-testid="toggle-button"><Icon name={icon} className={styles.sidebarIcon} /></span>
-              </span>
-              {
-                resultsEmpty
-                  ? <Spinner />
-                  : (
-                    <React.Fragment>
-                      <NavItem isSelected={open && selectedKey === 'viewDescription'}
-                               text="View Description"
-                               icon={<Icon name="info" />}
-                               onClick={this.setSelectedKey('viewDescription')}
-                               isLast={false}
-                               isOpen={open}>
-                        <React.Fragment>
-                          <div className={styles.viewMetadata}>
-                            <h3>{viewMetadata.title || defaultNewViewTitle}</h3>
-                            <small>{viewMetadata.summary || defaultNewViewSummary}</small>
-                          </div>
-
-                          <div className={styles.viewMetadata}>
-                            <SearchResultOverview results={results} />
-                          </div>
-                          {this.formatViewDescription(viewMetadata)}
-                        </React.Fragment>
-                      </NavItem>
-                      <NavItem isSelected={open && selectedKey === 'createWidget'}
-                               text="Create"
-                               icon={<Icon name="plus" />}
-                               onClick={this.setSelectedKey('createWidget')}
-                               isLast={false}
-                               isOpen={open}>
-                        <AddWidgetButton onClick={this.toggleOpen}
-                                         toggleAutoClose={this.toggleAutoClose}
-                                         queryId={queryId} />
-
-                      </NavItem>
-                      <NavItem isSelected={open && selectedKey === 'highlighting'}
-                               text="Formatting & Highlighting"
-                               icon={<Icon name="paragraph" />}
-                               onClick={this.setSelectedKey('highlighting')}
-                               isLast={false}
-                               isOpen={open}>
-                        <HighlightingRules />
-                      </NavItem>
-                      <NavItem isSelected={open && selectedKey === 'fields'}
-                               text="Fields"
-                               icon={<Icon name="subscript" />}
-                               onClick={this.setSelectedKey('fields')}
-                               isLast
-                               isOpen={open}>
-                        {children}
-                      </NavItem>
-                    </React.Fragment>
-                  )
-              }
-            </div>
-          </div>
-        </div>
-      </div>
+      <Container ref={(node) => { this.wrapperRef = node; }} open={open}>
```

Comment:
I still get the following floe error here:

Cannot assign `node` to `this.wrapperRef` because  mixed [1] is incompatible with  `HTMLDivElement`

It works, when I use the old way with `innerRef` but it throws many deprecated warnings, when running the tests.

Looks like this is related to styled-components, I could not find an elegant way.. the only "solution" which worked for me is adding the type `mixed` for the ref class attribute.

Reply:
Not a problem anymore.